string color
bool trigger
---
bool success